source/common/sink_impl.cc

```diff
@@ -0,0 +1,109 @@
+#include "common/sink_impl.h"
+
+#include <array>
+#include <filesystem>
+#include <fstream>
+
+#include "external/envoy/source/common/common/logger.h"
+#include "external/envoy/source/common/common/random_generator.h"
+
+namespace Nighthawk {
+namespace {
+
+absl::Status verifyStringIsGuid(absl::string_view s) {
+  Envoy::Random::RandomGeneratorImpl random;
```

Comment:
Well, my idea here is that the sink implementation will be the constraining factor when it comes to determining what is ok and not ok to specify for an execution_id. When initiating a load test, and not specifying an execution id, it will be auto-generated, and that will be a unique identifier.

In this file backed implementation I'm being a bit overly cautious, as a guid will have a reasonable length and contain no characters that would be illegal in a filename.  I hope to find a way to easily validate a filename and make sure there's no weird stuff in it.

Later on it will also be possible to manually specify an execution id when initiating an execution, and that might be useful when you want an id that is more easily remembered. As it stands, this implementation will reject that.